src/operators/Aggregate.cpp

```diff
@@ -0,0 +1,515 @@
+
+        status = schema_builder.AddFields(group_type->children());
+        evaluate_status(status, __FUNCTION__, __LINE__);
+
+        switch (kernel) {
+            // Returns a Datum of the same type INT64
+            case SUM: {
+                status = schema_builder.AddField(
+                        arrow::field(agg_col_name, arrow::int64()));
+                evaluate_status(status, __FUNCTION__, __LINE__);
+                break;
+            }
+            case MEAN: {
+                status = schema_builder.AddField(
+                        arrow::field(agg_col_name, arrow::float64()));
+                evaluate_status(status, __FUNCTION__, __LINE__);
+                break;
+            }
+            case COUNT: {
+                throw std::runtime_error("Count aggregate not supported.");
+            }
+        }
+
+        auto result = schema_builder.Finish();
+        evaluate_status(result.status(), __FUNCTION__, __LINE__);
+
+        return result.ValueOrDie();
+    }
+
+    std::shared_ptr<arrow::ChunkedArray> Aggregate::get_group_filter(int* its) {
+
+        arrow::Status status;
+        arrow::compute::FunctionContext function_context(
+                arrow::default_memory_pool());
+
+        // No Group By clause
+        if (group_type->num_children() == 0) {
+            return nullptr;
+        }
+
+        arrow::compute::Datum value;
+        std::shared_ptr<arrow::ChunkedArray> prev_filter;
+
+        // Fetch the next Group By filter and AND it with the previous filter
+        for (int field_i=0; field_i<group_type->num_children(); field_i++) {
+
+            std::shared_ptr<arrow::ChunkedArray> next_filter;
+
+            switch(group_type->child(field_i)->type()->id()) {
+                case arrow::Type::STRING: {
+                    auto one_unique_values_casted =
+                            std::static_pointer_cast<arrow::StringArray>
+                                    (unique_values_[field_i]);
+                    value = arrow::compute::Datum(
+                            std::make_shared<arrow::StringScalar>(
+                                    one_unique_values_casted->GetString(its[field_i])));
+                    next_filter = get_filter(group_by_refs_[field_i], value);
+                    break;
+                }
+                case arrow::Type::INT64: {
+                    auto one_unique_values_casted =
+                            std::static_pointer_cast<arrow::Int64Array>
+                                    (unique_values_[field_i]);
+                    value = arrow::compute::Datum(
+                            std::make_shared<arrow::Int64Scalar>(
+                                    one_unique_values_casted->Value(its[field_i])));
+                    next_filter = get_filter(group_by_refs_[field_i], value);
+                    break;
+                }
+                default: {
+                    std::cerr << "invalid type" << std::endl;
```

Comment:
This is good for this PR, but should we have a separate infrastructure to handle errors, so that we can use `throw` to catch the stack-trace of an error, and return line numbers. 

@johnchronis - Is there an error handling infrastructure you recommend?